gpumon/gpumon_interface.ml

```diff
@@ -0,0 +1,28 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+let service_name = "gpu"
+let queue_name = Xcp_service.common_prefix ^ service_name
+let json_path = "/var/xapi/" ^ service_name ^ ".json"
+let xml_path = "/var/xapi/" ^ service_name
+
+type debug_info = string
+
+type pgpu_address = string
+type pgpu_metadata = string
+type dom_id = int
+type vgpu_compatibility = unit (* temporarily unit, until real type is determined *)
+
+external get_pgpu_metadata: debug_info -> pgpu_address -> pgpu_metadata = ""
+external get_pgpu_vm_compatibility: debug_info -> pgpu_metadata -> dom_id -> vgpu_compatibility list = ""
```

Comment:
Oh, I thought we could get the metadata about the VM's (current) vGPUs from the domid without needing to specify which pgpu (or pgpus) the VM is currently using.